### PR TITLE
add randomness to one weight at time, new outputs and extra tests

### PR DESCRIPTION
**Pull request by Flaminietta**
Every single indicator can be associated with a randomly sampled weight from a uniform distribution for n-times; all the remaining weights are set equal to 1. MCDA is therefore run (_no_samples x no_indicators_) times. The mean and std of the final scores relative to every randomly sampled weight are estimated. This results in 8 different plots and outputs, each showing the variability related to the specific indicator examined. Results are saved in the output path.

### CODE REVIEW

File: mcda/utils.py [hunk 1]
```diff
@@ -163,6 +187,47 @@ def plot_mean_scores(all_weights_means:pd.DataFrame, all_weights_stds:pd.DataFra
     return fig
 
 
+def plot_mean_scores_iterative(all_weights_means:pd.DataFrame, all_weights_stds:pd.DataFrame, indicators: list, index: int)-> object:
+    no_of_combinations = all_weights_means.shape[1] - 1
```
Comment (kapil-agnihotri) [inline mcda/utils.py:191]: maybe `num_of_com...`?

File: tests/unit_tests/test_mcda_run.py [hunk 2]
```diff
@@ -31,22 +32,43 @@ def get_uncorrect_config_2():
...
                 "random_weights": "no",
+                "iterative": "no",
                 "no_samples": 10000,
                 "given_weights": [0.5, 0.5, 0.5, 0.5, 0.5, 0.5]
                                            },
             "output_path": "/path/to/output"
         }
 
+    @staticmethod
+    def get_uncorrect_config_3():
```
Comment (kapil-agnihotri) [inline tests/unit_tests/test_mcda_run.py:43]: Maybe `get_incorrect_config_*`?

**Review by kapil-agnihotri — APPROVED** [review]

**Review by mspada — APPROVED** [review]

**Review by mspada — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -26,8 +30,10 @@ The following input information are contained in the `configuration.json` file:
 - list of weights for each indicator 
     - should the weights be randomly sampled by mean of a Monte Carlo sampling? (*yes* or *no*)
     - if *no*, a list of weights should be given as input
+    - if *yes*, should the process be *iterative* (*yes*, i.e. randomness is associated to one weight at time) or not ("no", i.e. randomness is associated to all the weights at the same time)?
     - if *yes*, the number of samples should be given as input
-    - the sum of the weights should always be equal to 1 or the values will be rescaled
+    - the sum of the weights should always be equal to 1 or the values will be rescaled using minmax rescaler
```
[the remark this answers is not in the document]
Comment (Flaminietta) [reply README.md:35]: Good point, they should be normalized. I'm checking if _minmax rescaler_ is really normalizing them or just rescaling them.

[the remark this answers is not in the document]
Comment (Flaminietta) [reply README.md:35]: no minmax scaler is actually needed, all correct

File: README.md [hunk 4]
```diff
@@ -56,11 +62,19 @@ python3 -m pytest -s tests/unit_tests/test_mcda_run.py -vv
...
-If N=1 and the input matrix has no uncertainties associated to the indicators, then:
+If N=0 and the input matrix has no uncertainties associated to the indicators, then:
 - the indicator values are normalized by mean of all the possible normalization methods 
 - the normalized indicators are aggregated by mean of all the possible aggregation methods, by considering their assigned weights
-- results of all the combinations normalization/aggregation are provided
-If else, N>1 and the input matrix has uncertaities associate to some or all the indicators, then:
+- the results of all the combinations normalization/aggregation are provided
+
+If else N=0, the input matrix has no uncertainties associated to the indicators, but the weights are randomly sampled, then:
+- all weights or one weight at time are randomly sampled from a uniform distribution [0,1]
```
[the remark this answers is not in the document]
Comment (Flaminietta) [reply README.md:71]: yes, if randomness on the weight is _iterative_ or not

**Pull request merged**